source/bluetooth/MicroBitPartialFlashingService.cpp

```diff
@@ -329,17 +329,16 @@ void MicroBitPartialFlashingService::partialFlashingEvent(MicroBitEvent e)
       // Iterate through until reaching the scratch page
       while(flashPointer < (uint32_t *)DEFAULT_SCRATCH_PAGE)
       {
-        // Check for embedded source magic
-        if(*flashPointer == 0x41140EF && *(uint32_t *)(flashPointer + 0x1) == 0xB82FA2BB)
+
+        // Check if page is blank
+        if(crc32_compute((uint8_t*)flashPointer, MICROBIT_CODEPAGESIZE, NULL) != 0xc71c0011)
```

Comment:
Erases remaining pages in region for both MicroPython and MakeCode, so it will remove the embedded source if it is there

Reply:
I guess this means MakeCode partial flashing will now be slightly slower?

Reply:
Also, should the changes to this file be backported to the DAL as well?

Reply:
It's so quick that I don't think that we need to worry about it for MakeCode, doesn't make a noticable difference when I do it
Perhaps I ought to verify it's actually doing the erasure..

I don't think backporting to the DAL is necessary, we don't ever intend to have BLE + MicroPython on V1 do we?